average.js

```diff
@@ -1,6 +1,20 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  //return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+
+
+  var total = 0;// variable to store total sum of array:numbers
+  var count = 0;// total elements in array:numbers
+  var n = numbers.length;//length of array:numbers
+  for(let i=0;i<n;++i) {
+      if(isNaN(numbers[i])) continue;//check to skip NaN: not a Number
+      total += numbers[i];
```

Comment:
You are totalling and counting in a single pass. This is suitable in many situations.
However, consider the trade-off as well. If they are done in two loops, they could be done in parallel and be faster for large data-sets.